### PR TITLE
fix(api/auth): add OL_COOKIE_DOMAIN for split-subdomain deploys (#1725)

### LINKED ISSUE #1725: [BUG] Infrastructure — split-subdomain deploys log users out (auth cookies lack a Domain attribute)
## Problem / Context

When OpenLinker is deployed with the SPA and API on **different subdomains of the same registrable domain** (e.g. `app.demo.openlinker.io` for `apps/web` and `api.demo.openlinker.io` for `apps/api`), any **full-page navigation logs the user out** and bounces them to `/login`. The most visible trigger is the guided OAuth flow ("Connect with Allegro"), which does a full-page `window.location.assign(authorizationUrl)` (`apps/web/src/features/allegro/components/AllegroSetupForm.tsx`), but it applies to any full reload.

Root cause chain:

1. The access token is **in-memory only** (`apps/web/src/shared/auth/jwt-bearer-session-adapter.ts`, deliberate since #710). A full-page navigation destroys it, so re-authentication depends entirely on the silent `POST /v1/auth/refresh`.
2. `/v1/auth/refresh` is CSRF-protected (`apps/api/src/auth/guards/csrf.guard.ts`): the SPA must read the non-HttpOnly `ol_csrf` cookie via `document.cookie` and mirror it into the `x-csrf-token` header.
3. In `apps/api/src/auth/auth.cookies.ts`, both `ol_refresh` and `ol_csrf` are set **host-only** — no `Domain` attribute. So `ol_csrf` written by `api.demo.openlinker.io` is **not readable by `document.cookie` on `app.demo.openlinker.io`** (different host). `readCsrfCookie()` returns `null`, no `x-csrf-token` header is sent, and `CsrfGuard` rejects the refresh → session goes anonymous → `AuthenticatedAppLayout` redirects to `/login`.

`OL_COOKIE_SAMESITE` already exists (`resolveSameSite()`), so SameSite is configurable — but there is **no `Domain` knob**, which is the missing piece for split-subdomain deploys. The file's own comment (lines 45-52) anticipates the `app.example.com` / `api.example.com` topology but only solves the SameSite half.

This works fine in same-origin dev/single-host deploys, which is why it has not surfaced before; it breaks any split-subdomain production/demo topology.

## Proposed Solution

Add an `OL_COOKIE_DOMAIN` env var, read in `apps/api/src/auth/auth.cookies.ts`, and apply it as the `domain` option on **both** `refreshCookieOptions()` and `csrfCookieOptions()` when set. When unset, behaviour is unchanged (host-only cookies) — fully backward-compatible.

- Add a `resolveCookieDomain(): string | undefined` helper (mirrors `resolveSameSite()`): returns `process.env.OL_COOKIE_DOMAIN` trimmed, or `undefined` when empty/absent.
- Spread `...(domain ? { domain } : {})` into both cookie-option objects. Also pass `domain` on the matching `res.clearCookie(...)` calls in `clearAuthCookies()` / the migration-cleanup clears, since a cookie set with a `Domain` can only be cleared with the same `Domain`.
- Document the split-subdomain recipe: `OL_COOKIE_DOMAIN=.demo.openlinker.io` + `OL_COOKIE_SAMESITE=none` (or `lax`) + `NODE_ENV=production` (for `Secure`) + `OL_CORS_ORIGIN=https://app.demo.openlinker.io`. Update `apps/api/.env.example` and the relevant auth/deployment doc.

Follow the existing `resolveSameSite()` pattern exactly — env-driven, no code changes required per deploy, safe default.

Out of scope (separate FE follow-up, not required to fix the logout): the SPA callback route living under `AuthenticatedAppLayout` with a hard `/login` bounce and no return-to preservation. Fixing the cookie `Domain` restores silent refresh, which resolves the reported logout.

## Classification

**Type**: Infrastructure
**Layer**: Interface (auth cookie shaping)
**File(s)**:
- `apps/api/src/auth/auth.cookies.ts` (add `resolveCookieDomain()`, apply `domain` on set + clear)
- `apps/api/.env.example` (document `OL_COOKIE_DOMAIN`)
- auth/deployment doc — split-subdomain recipe

## Dependencies

- None. Builds on the existing `OL_COOKIE_SAMESITE` mechanism (already merged).

## Assumptions

- SPA and API share a registrable parent domain (e.g. both under `*.demo.openlinker.io`), so a `Domain=.demo.openlinker.io` cookie is valid for both. Fully cross-site (unrelated domains) is not supported by this approach and is not a target topology.
- `Secure` continues to be governed by `NODE_ENV=production`; `SameSite=none` requires it, which operators already control.

## Acceptance Criteria

- [ ] `OL_COOKIE_DOMAIN`, when set, applies as the `Domain` attribute on both `ol_refresh` and `ol_csrf` (set + clear paths).
- [ ] When `OL_COOKIE_DOMAIN` is unset/empty, cookies remain host-only (byte-identical to current behaviour).
- [ ] `clearAuthCookies()` and migration-cleanup clears pass the same `domain`, so cookies set with a Domain are actually cleared.
- [ ] Unit test in `apps/api/src/auth/auth.cookies.*spec.ts` (or a colocated spec) asserts: (a) domain present when env set, (b) absent when env unset, (c) applied to both cookies.
- [ ] `apps/api/.env.example` documents `OL_COOKIE_DOMAIN` and the split-subdomain recipe (`OL_COOKIE_DOMAIN` + `OL_COOKIE_SAMESITE=none` + `NODE_ENV=production` + `OL_CORS_ORIGIN`).
- [ ] `pnpm lint && pnpm type-check && pnpm test` pass.
- [ ] No architecture boundary violations.

### PR DESCRIPTION
**Pull request by norbert-kulus-blockydevs**
## What & why

Fixes a logout bug on **split-subdomain deploys** (SPA on `app.example.com`, API on `api.example.com`). Any full-page navigation — most visibly the guided "Connect with Allegro" OAuth redirect — bounced the user to `/login`.

Root cause chain:
1. The access token is in-memory only (#710), so a full-page navigation destroys it and re-auth depends entirely on the silent `POST /v1/auth/refresh`.
2. `/v1/auth/refresh` is CSRF-protected: the SPA reads the non-HttpOnly `ol_csrf` cookie via `document.cookie` and mirrors it into `x-csrf-token`.
3. In `apps/api/src/auth/auth.cookies.ts` both `ol_refresh` and `ol_csrf` were set **host-only** (no `Domain`). So `ol_csrf` written by `api.*` is unreadable from `app.*`, the CSRF header is empty, `CsrfGuard` rejects the refresh, and the session goes anonymous.

`OL_COOKIE_SAMESITE` already existed; the missing piece was a `Domain` knob.

## Change

- **`auth.cookies.ts`** — new `resolveCookieDomain()` + `domainOption()` (mirrors `resolveSameSite()`). When `OL_COOKIE_DOMAIN` is set, its value is applied as the `Domain` attribute on both cookies on set **and** on the current-path clears in `clearAuthCookies` (a Domain-scoped cookie only clears with the same Domain). Legacy `/auth` clears stay host-only — those copies predate this change and were never Domain-scoped.
- **`auth.cookies.spec.ts`** — new unit spec: domain applied when set, absent when unset (host-only, unchanged), applied to both cookies + current clears but not legacy clears, value trimmed, whitespace-only treated as unset.
- **`apps/api/.env.example`** — documents `OL_COOKIE_DOMAIN` + `OL_COOKIE_SAMESITE` and the split-subdomain recipe.

**Backward-compatible:** with `OL_COOKIE_DOMAIN` unset, cookie shaping is byte-identical to before.

## Split-subdomain recipe

```
OL_COOKIE_DOMAIN=.example.com
OL_COOKIE_SAMESITE=none        # cross-origin XHR; requires Secure
NODE_ENV=production            # turns on Secure (needs real TLS)
OL_CORS_ORIGIN=https://app.example.com
```

## Out of scope

The FE callback route sitting under `AuthenticatedAppLayout` with a hard `/login` bounce and no return-to preservation is a separate follow-up. Fixing the cookie `Domain` restores silent refresh, which resolves the reported logout.

## Testing

- `apps/api` unit spec: 8/8 pass
- `pnpm --filter @openlinker/api type-check` — pass
- lint on changed files — pass

Closes #1725

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/api/src/auth/auth.cookies.ts [hunk 1]
```diff
@@ -56,13 +56,34 @@ function resolveSameSite(): 'strict' | 'lax' | 'none' {
...
+// app.example.com, API on api.example.com). Without a Domain the cookies are
+// host-only, so the SPA host can't read the non-HttpOnly ol_csrf cookie set by
+// the API host via document.cookie — the X-CSRF-Token mirror comes up empty and
+// every silent /auth/refresh after a full-page navigation (e.g. the OAuth
+// bounce back from allegro.pl) is rejected, logging the user out. Setting
+// OL_COOKIE_DOMAIN=.example.com scopes both cookies to the shared parent domain
+// so the SPA subdomain can read ol_csrf. Unset ⇒ host-only (unchanged). See
+// #1725. A Domain-scoped cookie can only be cleared with the same Domain, so
+// this value MUST also be threaded through every clearCookie() below.
+function resolveCookieDomain(): string | undefined {
```
Comment (piotrswierzy) [inline apps/api/src/auth/auth.cookies.ts:69]: **SUGGESTION (security caveat, non-blocking).** The mechanics here are correct: a leading-dot `Domain=.example.com` is exactly what the CSRF-mirror flow needs, and defaulting to `undefined` (host-only) is the right safe default. Worth surfacing the tradeoff explicitly though — scoping both cookies to the registrable parent broadens **both** the HttpOnly `ol_refresh` and the non-HttpOnly `ol_csrf` to *every* subdomain under it. On a deploy where an untrusted/lower-trust subdomain exists (user-content, a staging box, a third-party-hosted subdomain), that subdomain can read `ol_csrf` via `document.cookie` and receives `ol_refresh` on its requests. This is inherent to the feature and fully operator-controlled, so no code change needed — but a one-line "only set this when every subdomain under the parent is trusted" caveat in the `.env.example` block would keep an operator from over-scoping. Consider also rejecting/warning on obviously-wrong values (embedded `://`, path, or port) since `res.cookie()` will otherwise throw a raw `TypeError` at first request rather than failing loudly at boot.

File: apps/api/src/auth/auth.cookies.ts [hunk 2]
```diff
@@ -104,8 +126,12 @@ export function setCsrfCookie(res: Response): string {
...
 
 export function clearAuthCookies(res: Response): void {
-  res.clearCookie(REFRESH_COOKIE_NAME, { path: REFRESH_COOKIE_PATH });
-  res.clearCookie(CSRF_COOKIE_NAME, { path: CSRF_COOKIE_PATH });
+  // Current cookies may carry a Domain (OL_COOKIE_DOMAIN); a Domain-scoped
+  // cookie only clears when the clear carries the same Domain. The legacy
+  // /auth clears below stay host-only — those copies predate #1725 and were
+  // never Domain-scoped.
+  res.clearCookie(REFRESH_COOKIE_NAME, { path: REFRESH_COOKIE_PATH, ...domainOption() });
+  res.clearCookie(CSRF_COOKIE_NAME, { path: CSRF_COOKIE_PATH, ...domainOption() });
```
Comment (piotrswierzy) [inline apps/api/src/auth/auth.cookies.ts:134]: **SUGGESTION (reconfiguration edge case, non-blocking).** Threading `domainOption()` through the current-path clears is correct and the test covers it. One residual edge, in the same spirit as the legacy-path cleanup you're already handling: if an operator *changes* `OL_COOKIE_DOMAIN` after cookies have been issued (host-only → domain-scoped, or vice-versa), `clearAuthCookies` now only clears the variant matching the *current* config, leaving the previously-issued variant lingering. Because both share name + path `/`, the stale copy can shadow the new one on the next `/v1/auth/refresh` (same class of ol_csrf double-cookie shadowing bug the #748/#1327 cleanups fight). This is a rare one-time operator action and out of scope for the reported logout fix — not asking you to solve it here — but a follow-up that clears both the domain-scoped and host-only variant on logout (mirroring the legacy-path belt-and-braces already in this function) would close it.

**Review by piotrswierzy — APPROVED** [review]
## Tech Lead Review — PR #1726 (`OL_COOKIE_DOMAIN` for split-subdomain deploys)

**Verdict: ✅ APPROVE** — clean, correctly scoped, backward-compatible, well-tested, and security-sound. Two non-blocking suggestions inline; neither needs to hold the merge.

### Summary
Adds an `OL_COOKIE_DOMAIN` env knob that stamps a `Domain` attribute on both `ol_refresh` and `ol_csrf` when set, unblocking the silent `/v1/auth/refresh` on split-subdomain deploys (SPA on `app.*`, API on `api.*`). Root-cause chain in the issue/PR body is accurate — host-only `ol_csrf` is unreadable cross-subdomain, the `x-csrf-token` mirror comes up empty, `CsrfGuard` 403s, session goes anonymous. The `Domain` knob is the right minimal fix and mirrors the existing `resolveSameSite()` pattern in the same file.

### Security review (this is the load-bearing part)
- **Safe default ✅** — `resolveCookieDomain()` returns `undefined` on unset / empty / whitespace-only; `domainOption()` then spreads nothing, so with the var unset the cookie shaping is byte-identical to before (`...undefined` is a no-op in an object literal). Confirmed by the unset-path tests.
- **Secure / HttpOnly / SameSite preserved ✅** — `domainOption()` only ever contributes a `domain` key spread onto the existing options object; `httpOnly`, `secure: isProd()`, `sameSite: resolveSameSite()`, `path`, and `maxAge` are all untouched on both cookies.
- **Clear correctness ✅** — a `Domain`-scoped cookie only clears with a matching `Domain`; the value is correctly threaded through the current-path clears in `clearAuthCookies`, while the legacy `/auth` migration clears correctly stay host-only (those copies were never domain-scoped). Test asserts exactly this split.
- **Over-scoping caveat (inline, non-blocking)** — the leading-dot `Domain` necessarily broadens *both* the HttpOnly refresh token and the readable CSRF cookie to every subdomain of the parent. Inherent to the feature and operator-controlled; a trust caveat in `.env.example` + light value-validation would harden it. See inline.
- **Reconfiguration edge (inline, non-blocking)** — changing the var after cookies are issued can leave a stale host-only/domain-scoped sibling that shadows on `/refresh`; same class as the #748/#1327 shadowing bugs. Rare one-time operator action, reasonable follow-up. See inline.

### Architecture / standards
- No hexagonal boundary concerns — this is `apps/api` interface-layer cookie shaping. Reading `process.env` directly is consistent with the existing `isProd()` / `resolveSameSite()` in the same module.
- Types clean — `Pick<CookieOptions, 'domain'> | undefined`, no `any`. `CookieOptions` already imported.
- Tests are thorough: set/unset for both cookies, current-vs-legacy clear split, trim, whitespace-only-as-unset (8/8). Minor optional add: a regression assertion that `secure`/`httpOnly`/`sameSite` still co-exist alongside `domain`.
- `.env.example` documents both the knob and the full split-subdomain recipe (`SAMESITE=none` + `Secure`/TLS + CORS origin) — good operator ergonomics.

Nice, tightly-scoped fix with genuinely load-bearing comments explaining the RFC 6265 reasoning. Approving.
